script/run-unit-test.js

```diff
@@ -44,12 +49,83 @@ const testRunner = options.coverage ? coveragePath : mochaPath;
 const mochaOpts =
   'src/platform/testing/unit/mocha.opts src/platform/testing/unit/helper.js';
 
-// Otherwise, run the command
-runCommand(
-  `LOG_LEVEL=${options[
-    'log-level'
-  ].toLowerCase()} ${testRunner} --max-old-space-size=4096 --opts ${mochaOpts} --recursive ${options.path
-    .map(p => `'${p}'`)
-    .join(' ')}`,
-  options.coverage ? null : 0,
-);
+const logLevel = options['log-level'].toLowerCase();
+const forcedExitCode = options.coverage ? null : 0;
+const testDirectories = options.path;
+
+if (testDirectories[0] === defaultPath) {
+  const allUnitTests = glob.sync(defaultPath);
+
+  const unitTestDirectories = new Set(
```

Comment:
```suggestion
  // Group unit tests into suites by directories by creating a list of all unit test containing directories from the list of allUnitTestPaths
  const unitTestDirectories = new Set(
```
update if not accurate or if you want to change the wording. i think it would be helpful to describe why we're doing this- for me it's not necessary to understand the directory list munging to know what this is doing.